statsmodels/distributions/discrete.py

```diff
@@ -22,3 +22,22 @@ def _pmf(self, x, mu, alpha, p):
 
 genpoisson_p = genpoisson_p_gen(name='genpoisson_p',
                                 longname='Generalized Poisson')
+
+class zipoisson_gen(rv_discrete):
+    '''A Zero Inflated Poisson distribution
+    '''
+    def _argcheck(self, mu, w):
+        return (mu > 0) & (w >= 0) & (w<=1)
+
+    def _logpmf(self, x, mu, w):
+        if x == 0:
```

Comment:
This needs to work vectorized, scipy has a lazy_where function that is used in the scipy distributions